neo/Network/P2P/Payloads/GetFullBlocksPayload.cs

```diff
@@ -0,0 +1,37 @@
+using Neo.IO;
+using System;
+using System.IO;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class GetFullBlocksPayload : ISerializable
+    {
+        public const ushort MaxBlocksCount = 500;
```

Comment:
Also we need a `MaxSize`, my concern is that attackers could overflow the network asking for big packets to anyone. Now, we have 33MB per packet (https://github.com/neo-project/neo/blob/4f02c1ee7254e90cdb73b3b57a924811548d780c/neo/Network/P2P/Message.cs#L12) but i prefer a mechanism like https://github.com/neo-project/neo/pull/1088/files#diff-db6615aadf426a61f06d9a0880c3ad95R11-R12 for prevent this kind of situations

Reply:
Maybe you can create multiple message for do the same.... :S

Reply:
> Also we need a `MaxSize`, my concern is that attackers could overflow the network asking for big packets to anyone. Now, we have 33MB per packet

Fair observation, but that problem also exists without this PR. This existing handler for `GetBlocks`
https://github.com/neo-project/neo/blob/1c84f7cf9f211f386107677ce28b26de26042677/neo/Network/P2P/ProtocolHandler.cs#L158-L160

does the same as the newly proposed but is based on hashes instead of indices. So the only difference is that attackers first need to collect `current chain height / MaxBlocksCount ` hashes (about ~9000 on 2.x mainnet) and then they can request the same information.

I think we need to address this 33MB limit in a separate issue/PR.

Reply:
Agree